specification/cognitiveservices/data-plane/TranslatorText/preview/v1.0-preview.1/TranslatorBatch.json

```diff
@@ -958,11 +989,138 @@
         "Succeeded",
         "Failed",
         "Cancelled",
-        "Cancelling"
+        "Cancelling",
+        "ValidationFailed"
       ],
-      "type": "string"
+      "type": "string",
+      "x-ms-enum": {
+        "name": "Status",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "NotStarted"
+          },
+          {
+            "value": "Running"
+          },
+          {
+            "value": "Succeeded"
+          },
+          {
+            "value": "Failed"
+          },
+          {
+            "value": "Cancelled"
+          },
+          {
+            "value": "Cancelling"
+          },
+          {
+            "value": "ValidationFailed"
+          }
+        ]
+      }
+    },
+    "ErrorCodeV2": {
+      "description": "Enums containing high level error codes.",
+      "enum": [
+        "InvalidRequest",
+        "InvalidArgument",
+        "InternalServerError",
+        "ServiceUnavailable",
+        "ResourceNotFound",
+        "Unauthorized",
+        "RequestRateTooHigh"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ErrorCodeV2",
```

Comment:
This will generate an enum type named `ErrorCodeV2` however you already define a model with the same name (the model that contains this enum).  Autorest can deduplicate the name however your left to the whims of its algorithm.  Suggest you change one of the names.

Reply:
@PhoenixHe-msft why doesn't the linter complain about this?

Reply:
@jianyexi to have a look. Do we have lint rule about this?

Reply:
Not yet, are we really needed to enforce  the x-ms-enum name different with the  enum model name ?

Reply:
Well, you can't have `class ErrorCodeV2` and `enum ErrorCodeV2` defined in the same code.  As I said earlier, autorest will go ahead and disambiguate the name for you, however if you'd like to control the exact name yourself you should disambiguate them here.

Reply:
wait, so what's the pattern i should follow?

Should i just rename   "x-ms-enum": {
        "name": "ErrorCodeV2",

to say ErrorCodeV2Enum ?

The rest of the swagger should still reference ErrorCodeV2 right?

Reply:
Sigh please disregard, I misread this.  `ErrorCodeV2` is only used to define the enum type.